core/invoker/src/main/scala/org/apache/openwhisk/core/invoker/FPCInvokerReactive.scala

```diff
@@ -0,0 +1,483 @@
+                                 rpcPort: Int,
+                                 containerId: ContainerId): ActorRef = {
+    implicit val transId = TransactionId.invokerNanny
+    f.actorOf(
+      ActivationClientProxy
+        .props(invocationNamespace, fqn, rev, schedulerHost, rpcPort, containerId, activationClientFactory(etcdClient)))
+  }
+
+  val prewarmingConfigs: List[PrewarmingConfig] = {
+    ExecManifest.runtimesManifest.stemcells.flatMap {
+      case (mf, cells) =>
+        cells.map { cell =>
+          PrewarmingConfig(cell.initialCount, new CodeExecAsString(mf, "", None), cell.memory)
+        }
+    }.toList
+  }
+
+  private val pool =
+    actorSystem.actorOf(
+      ContainerPoolV2
+        .props(childFactory, invokerHealthManager, poolConfig, instance, prewarmingConfigs, sendAckToScheduler))
+
+  private def getLiveContainerCount(invocationNamespace: String,
+                                    fqn: FullyQualifiedEntityName,
+                                    revision: DocRevision): Future[Long] = {
+    val namespacePrefix = containerPrefix(ContainerKeys.namespacePrefix, invocationNamespace, fqn, Some(revision))
+    val inProgressPrefix = containerPrefix(ContainerKeys.inProgressPrefix, invocationNamespace, fqn, Some(revision))
+    val warmedPrefix = containerPrefix(ContainerKeys.warmedPrefix, invocationNamespace, fqn, Some(revision))
+    for {
+      namespaceCount <- etcdClient.getCount(namespacePrefix)
+      inProgressCount <- etcdClient.getCount(inProgressPrefix)
+      warmedCount <- etcdClient.getCount(warmedPrefix)
+    } yield {
+      namespaceCount + inProgressCount + warmedCount
+    }
+  }
+
+  /** Initialize message consumers */
+  private val msgProvider = SpiLoader.get[MessagingProvider]
+  //number of peeked messages - increasing the concurrentPeekFactor improves concurrent usage, but adds risk for message loss in case of crash
+  private val maxPeek = loadConfigOrThrow[Int](ConfigKeys.containerCreationMaxPeek)
+  private var consumer: Option[ContainerMessageConsumer] = Some(
+    new ContainerMessageConsumer(
+      instance,
+      pool,
+      entityStore,
+      cfg,
+      msgProvider,
+      longPollDuration = 1.second,
+      maxPeek,
+      sendAckToScheduler))
+
+  override def enable(): Route = {
+    invokerHealthManager ! Enable
+    pool ! Enable
+    // re-enable consumer
+    if (consumer.isEmpty)
+      consumer = Some(
+        new ContainerMessageConsumer(
+          instance,
+          pool,
+          entityStore,
+          cfg,
+          msgProvider,
+          longPollDuration = 1.second,
+          maxPeek,
+          sendAckToScheduler))
+    warmUp()
+    complete("Success enable invoker")
+  }
+
+  override def disable(): Route = {
+    invokerHealthManager ! GracefulShutdown
+    pool ! GracefulShutdown
+    consumer.foreach(_.close())
+    consumer = None
+    warmUpWatcher.foreach(_.close())
+    warmUpWatcher = None
+    complete("Success disable invoker")
```

Comment:
nit: `Successfully disabled invoker`. Is it possible to include the invoker id here?

Reply:
Updated accordingly